libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -1064,19 +1068,23 @@ string YulUtilFunctions::clearStorageRangeFunction(Type const& _type)
 {
 	string functionName = "clear_storage_range_" + _type.identifier();
 
-	solAssert(_type.storageBytes() >= 32, "Expected smaller value for storage bytes");
-
 	return m_functionCollector.createFunction(functionName, [&]() {
 		return Whiskers(R"(
-			function <functionName>(start, end) {
+			function <functionName>(start, offset, end) {
+				if gt(offset, 0) {
+					let mask := <shr>(<ones>, mul(8, sub(32, offset)))
+					sstore(start, and(mask, sload(start)))
```

Comment:
Not exactly sure if this can be tested before #9738. If there is no way to resize storage array to some length than I am not able to test it right now.